arm/ripper/utils.py

```diff
@@ -194,6 +196,53 @@ def make_dir(path):
     else:
         return False
 
+def detect_disctype(devpath):
+    """Parse udev for properties of current disc"""
+    disctype = "unknown"
+    label = "unknown"
+    context = pyudev.Context()
+    device = pyudev.Devices.from_device_file(context, devpath)
+    return parse_udev(device.items())
+
+def parse_udev_cmdline(args):
+    udev = {}
+    if args.disctype:
+        (k, v) = args.disctype.split('=', 1)
+        udev[k] = v
+    if args.label:
+        (k, v) = args.label.split('=', 1)
+        udev[k] = v
+    return parse_udev(udev)
+
+def parse_udev(udev_dict):
+    map_udev_disctype = {
+            "ID_CDROM_MEDIA_BD": "bluray",
+            "ID_CDROM_MEDIA_DVD": "dvd",
+            "ID_CDROM_MEDIA_TRACK_COUNT_AUDIO": "music"
+            }
+    disctype == "unknown"
+    label = udev_dict.get("ID_FS_LABEL", None)
+    if label is None:
+        label == "unknown"
+    elif label == "iso9660":
+        disctype = "data"
+    else:
+        # check for known disctypes
+        for key in map_udev_disctype.keys():
```

Comment:
I couldn't get Music discs to ever hit this code. I had to rewrite this to get it working with music CD's.

This also isnt helpful for debugging should anything ever go wrong. As it outputs nothing to the user.